astropy/modeling/tests/test_compound.py

```diff
@@ -543,175 +282,87 @@ def test_identity_input():
-    p1 = Polynomial2D(1, c0_0=1, c1_0=2, c0_1=3)
-    mapping = Mapping((0, 1, 0, 1))
-    offx = Shift(-2, name='x_translation')
-    offy = Shift(-1, name='y_translation')
-    aff = AffineTransformation2D(matrix=[[1, 2], [3, 4]], name='rotation')
-    model = mapping | (p1 & p2) | (offx & offy) | aff
-
-    assert model.param_names == ('c0_0_1', 'c1_0_1', 'c0_1_1',
-                                 'c0_0_2', 'c1_0_2', 'c0_1_2',
-                                 'offset_3', 'offset_4',
-                                 'matrix_5', 'translation_5')
-    assert model(1, 2) == (23.0, 53.0)
-
-    m = model[3:]
-    assert m.param_names == ('offset_3', 'offset_4', 'matrix_5',
-                             'translation_5')
-    assert m(1, 2) == (1.0, 1.0)
+        Rotation2D(90) + Gaussian1D(1, 0, 0.1)
 
 
-def test_compound_model_with_nonstandard_broadcasting():
+@pytest.mark.parametrize('poly', [Chebyshev2D(1, 2), Polynomial2D(2), Legendre2D(1, 2),
+                                  Chebyshev1D(5), Legendre1D(5), Polynomial1D(5)])
+def test_compound_with_polynomials(poly):
     """
-    Ensure that the ``standard_broadcasting`` flag is properly propagated when
-    creating compound models.
-
-    See the commit message for the commit in which this was added for more
-    details.
+    Tests that polynomials are scaled when used in compound models.
+    Issue #3699
     """
+    poly.parameters = [1, 2, 3, 4, 1, 2]
+    shift = Shift(3)
+    model = poly | shift
+    x, y = np.mgrid[:20, :37]
+    result_compound = model(x, y)
+    result = shift(poly(x, y))
+    assert_allclose(result, result_compound)
 
-    offx = Shift(1)
-    offy = Shift(2)
-    rot = AffineTransformation2D([[0, -1], [1, 0]])
-    m = (offx & offy) | rot
-
-    x, y = m(0, 0)
-    assert x == -2
-    assert y == 1
-
-    # make sure conversion back to scalars is working properly
-    assert isinstance(x, float)
-    assert isinstance(y, float)
-
-    x, y = m([0, 1, 2], [0, 1, 2])
-    assert np.all(x == [-2, -3, -4])
-    assert np.all(y == [1, 2, 3])
-
-
-def test_compound_model_classify_attributes():
-    """
-    Regression test for an issue raised here:
-    https://github.com/astropy/astropy/pull/3231#discussion_r22221123
 
-    The issue is that part of the `help` implementation calls a utility
-    function called `inspect.classify_class_attrs`, which was leading to an
-    infinite recursion.
+# def test_variable_substitution():
+#     g1 = Gaussian2D(1, 0, 0, 1, 2)
+#     g2 = Gaussian2D(1.5, .5, -.2, .5, .3)
+#     sg1_1 = g1 % {1: 0}
+#     assert_almost_equal(sg1_1(0), g1(0, 0))
+#     sg1_2 = g1 % {'x': 1}
+#     assert_almost_equal(sg1_2(1.5), g1(1, 1.5))
+#     gg1 = g1 & g2
+#     sgg1_1 = gg1 % {1: 0.1, 3: 0.2}
+#     assert_almost_equal(sgg1_1(0, 0), gg1(0, 0.1, 0, 0.2))
+#     sgg1_2 = gg1 % {'x0': -.1, 2: .1}
+#     assert_almost_equal(sgg1_2(1,1), gg1(-0.1, 1, 0.1, 1))
+#     with pytest.raises(ValueError):
+#         egg1 = gg1 % {'x0': 0, 0: 0}
```

Comment:
Just wondering whether this and other commented-out tests are supposed to run eventually or not? If they are no longer useful, they should be removed entirely. If they don't pass yet, it might be useful to see the failures on CI.

Reply:
This segment was waiting for the variable substitution code to be layered onto this PR. I've removed in this branch since it has been uncommented in that branch